ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/tree/nodes/pairs/values/TomlLong.kt

```diff
@@ -1,22 +1,53 @@
 package com.akuleshov7.ktoml.tree.nodes.pairs.values
 
 import com.akuleshov7.ktoml.TomlOutputConfig
+import com.akuleshov7.ktoml.writers.IntegerRepresentation
+import com.akuleshov7.ktoml.writers.IntegerRepresentation.*
 import com.akuleshov7.ktoml.writers.TomlEmitter
 
 /**
  * Toml AST Node for a representation of Arbitrary 64-bit signed integers: key = 1
  * @property content
+ * @property representation The representation of the integer.
  */
 public class TomlLong internal constructor(
-    override var content: Any
+    override var content: Any,
+    public var representation: IntegerRepresentation = DECIMAL
 ) : TomlValue() {
-    public constructor(content: String, lineNo: Int) : this(content.toLong())
+    public constructor(content: String, lineNo: Int) : this(content.parse())
```

Comment:
Actually I do not get it - if you remember I was always avoiding custom logic for parsing and even made logic for parsing based on exceptions: 
```
try {
    s.toLong
} catch() {
    try {
        s.toInt
    }
}

...
etc.
```

So I tried as much as possible to avoid custom parsing. Why do we need to have it here?

Reply:
Hmm, I did it this way to support parsing the other integer types, instead of just decimal. `toLong` only parses decimal numbers, it can't handle the the prefixes and non-decimal bases that TOML supports, like `0b`, `0x`, etc. We can do it a different way if you have something in mind

Reply:
I thought that this parsing of prefixes is included into `toLong`… I got the intention now